plugin.cpp

```diff
@@ -248,6 +253,55 @@ PLUGIN_HANDLE plugin_init(ConfigCategory* config,
 	return (PLUGIN_HANDLE)info;
 }
 
+
+/**
+ * Flatten nested datapoint
+ *
+ * @param datapoint	datapoint to flatten
+ * @param assetName	assetname to which datapoint belongs
+ * @param newReadings	vector for newreadings of flatten datapoint
+ */
+void flattenDatapoint(Datapoint *datapoint, std::string assetName, vector<Reading *>& newReadings)
+{
+	static std::string datapointName = datapoint->getName();
+	DatapointValue datapointValue = datapoint->getData();
+
+	std::vector<Datapoint *> *&nestedDP = datapointValue.getDpVec();
+
+	auto dpit = nestedDP->begin();
+	for (auto dpit = nestedDP->begin(); dpit != nestedDP->end(); dpit++)
+	{
+
+		std::string name = (*dpit)->getName();
+		DatapointValue &val = (*dpit)->getData();
+		if (val.getType() == DatapointValue::T_STRING)
+		{
+			name = datapointName + "_" + name;
+			newReadings.emplace_back(new Reading(assetName , new Datapoint(name, val)) );
```

Comment:
You should not be creating a new reading for every datapoint, the datapoints from a single source reading all end up in a single destination reading, just with the datapoints flattened.